pkg/ec2metadata/ec2metadata.go

```diff
@@ -104,15 +104,16 @@ type RebalanceRecommendation struct {
 
 // NodeMetadata contains information that applies to every drain event
 type NodeMetadata struct {
-	AccountId        string `json:"accountId"`
-	InstanceID       string `json:"instanceId"`
-	InstanceType     string `json:"instanceType"`
-	PublicHostname   string `json:"publicHostname"`
-	PublicIP         string `json:"publicIp"`
-	LocalHostname    string `json:"localHostname"`
-	LocalIP          string `json:"privateIp"`
-	AvailabilityZone string `json:"availabilityZone"`
-	Region           string `json:"region"`
+	AccountId        string            `json:"accountId"`
+	InstanceID       string            `json:"instanceId"`
+	InstanceType     string            `json:"instanceType"`
+	PublicHostname   string            `json:"publicHostname"`
+	PublicIP         string            `json:"publicIp"`
+	LocalHostname    string            `json:"localHostname"`
+	LocalIP          string            `json:"privateIp"`
+	AvailabilityZone string            `json:"availabilityZone"`
+	Region           string            `json:"region"`
+	NodeLabels       map[string]string `json:"nodeLabels"`
```

Comment:
Yeah, I gave it a thought while putting it here.

Since the name of struct is `NodeMetadata` (node labels kinda go with node metadata) and it has nothing to do with interruptionEvent. I looked at `interruptioneventstore` and it didn't really fit there well. I'm still not fully convinced of adding it there. Since you are more familiar with the package setup, I'll move it there if you still feel so :) Let me know your thoughts.

Reply:
I agree it's not a perfect fit with the InterruptionEvent, however, the bigger issue is the pkg ec2metadata. The pkg is really what's defining the API and it's assumed that the data will be coming from ec2metadata, not k8s API server. 

A better option might be to add a K8s node struct to the webhook combinedData struct to store all the relevant k8s node data.  I thought InterruptionEvent is okay since it includes the NodeName already... but it's honestly not a great reason.